### PR TITLE
r/opc_storage_object: Add `opc_storage_object` resource

### PR DESCRIPTION
**Pull request by grubernaut**
Still need to add timeouts to the resource, but those can be added later.
Also need to add Import and Import test

```
$ make testacc TEST=./opc TESTARGS="-run=TestAccOPCStorageObject"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./opc -v -run=TestAccOPCStorageObject -timeout 120m
=== RUN   TestAccOPCStorageObject_contentSource
--- PASS: TestAccOPCStorageObject_contentSource (17.08s)
=== RUN   TestAccOPCStorageObject_fileSource
--- PASS: TestAccOPCStorageObject_fileSource (12.49s)
PASS
ok      github.com/terraform-providers/terraform-provider-opc/opc       29.577s
```

### CODE REVIEW

File: opc/resource_storage_object.go [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Set the content-encoding metadata",
+			},
+			"content_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Set the MIME type for the object",
```
Comment (tombuildsstuff) [inline opc/resource_storage_object.go]: Do we need a default here - or does the API handle that for us?

File: opc/resource_storage_object.go [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+					}
+					return
+				},
+			},
+			"delete_at": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "Specify the number of seconds after which the system deletes the object",
```
Comment (tombuildsstuff) [inline opc/resource_storage_object.go]: > number of seconds

I don't know the use-case for this, but would it make sense to put this in a larger unit? We also have this in the disk sizes, which are submitted in Bytes but are either MB/GB in the Schema

File: opc/resource_storage_object.go [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+		file, err := os.Open(path)
+		if err != nil {
+			return fmt.Errorf("Error opening Storage Object file (%s): %s", source, err)
+		}
+		input.Body = file
+	} else if v, ok := d.GetOk("copy_from"); ok {
+		input.CopyFrom = v.(string)
+	} else {
+		// One of the three attributes are required
+		return fmt.Errorf("Must specify %q, %q, or %q field", "file", "copy_from", "content")
```
Comment (tombuildsstuff) [inline opc/resource_storage_object.go]: do these field names want ` quotes around them?

File: opc/resource_storage_object.go [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+	d.Set("timestamp", result.Timestamp)
+	d.Set("transaction_id", result.TransactionID)
+
+	return nil
+}
+
+func resourceOPCStorageObjectDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*OPCClient).storageClient.Objects()
+	if client == nil {
+		return fmt.Errorf("Storage client is not initialized. Make sure to use `storage_endpoint` variable or the `OPC_STORAGE_ENDPOINT` environment variable")
```
Comment (tombuildsstuff) [inline opc/resource_storage_object.go]: _minor:_ is it worth making this error message a constant?

File: opc/resource_storage_object_test.go [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+	"github.com/hashicorp/go-oracle-terraform/storage"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+const _TestStorageObjectPath = "test-fixtures"
+
+func TestAccOPCStorageObject_contentSource(t *testing.T) {
```
Comment (tombuildsstuff) [inline opc/resource_storage_object_test.go]: Given fields like `content_type` are Optional but always supplied in the test - it'd be good to add a "basic" test with only the required fields set if possible?

File: opc/resource_storage_object.go (hunk 1 above)
Comment (grubernaut) [inline opc/resource_storage_object.go]: API handles this for us if left unspecified

File: opc/resource_storage_object.go (hunk 3 above)
Comment (grubernaut) [inline opc/resource_storage_object.go]: Sure, can use those instead

File: opc/resource_storage_object.go (hunk 4 above)
Comment (grubernaut) [inline opc/resource_storage_object.go]: Absolutely :smile:

File: opc/resource_storage_object_test.go (hunk 5 above)
Comment (grubernaut) [inline opc/resource_storage_object_test.go]: Aye, agreed. Will add

File: opc/resource_storage_object.go (hunk 2 above)
Comment (grubernaut) [inline opc/resource_storage_object.go]: Possibly... I'd almost opt for adding an interpolation function to `core` that converts hours to minutes, and one that converts minutes to seconds, so a user can fully specify seconds if needed.

Comment (scross01) [inline opc/resource_storage_object.go]: I like to idea of an interpolation function for this

**Pull request merged**